src/main/java/org/broadinstitute/hellbender/tools/exome/copyratio/CopyRatioData.java

```diff
@@ -23,40 +26,35 @@
     private final int numTargets;
     private final double coverageMin;
     private final double coverageMax;
-    private final double coverageRange;
 
-    private final List<List<Double>> coveragesPerSegment = new ArrayList<>();
-    private final List<Integer> numTargetsPerSegment;
-    private final List<Integer> startTargetsPerSegment = new ArrayList<>();
+    private final List<List<IndexedCoverage>> indexedCoveragesPerSegment = new ArrayList<>();
 
     public CopyRatioData(final SegmentedGenome segmentedGenome) {
-        if (segmentedGenome.getGenome().getTargets().targetCount() == 0) {
-            throw new IllegalArgumentException("Cannot construct CopyRatioData with no target-coverage data.");
-        }
-        //construct coverages and number of targets per segment from the segments and Genome in SegmentedGenome
         final TargetCollection<ReadCountRecord.SingleSampleRecord> targetCoverages = segmentedGenome.getGenome().getTargets();
+        Utils.validateArg(targetCoverages.targetCount() > 0, "Cannot construct CopyRatioData with no target-coverage data.");
         //construct list of coverages (in order corresponding to that of segments in SegmentedGenome;
         //this may not be in genomic order, depending on how the segments are sorted in the segment file,
         //so we cannot simply take the list of coverages in the order from TargetCollection.targets()
-        final List<Double> coverages = segmentedGenome.getSegments().stream()
+        final List<SimpleInterval> segments = segmentedGenome.getSegments();
+        numSegments = segments.size();
+        final List<Double> coverages = segments.stream()
                 .flatMap(s -> targetCoverages.targets(s).stream())
                 .map(ReadCountRecord.SingleSampleRecord::getCount)
                 .collect(Collectors.toList());
-        numTargetsPerSegment = segmentedGenome.getSegments().stream()
-                .map(s -> targetCoverages.targets(s).size())
-                .collect(Collectors.toList());
-        numSegments = numTargetsPerSegment.size();
         numTargets = coverages.size();
         coverageMin = Collections.min(coverages);
         coverageMax = Collections.max(coverages);
-        coverageRange = coverageMax - coverageMin;
-        //partition coverages by segment
-        int startTarget = 0;
-        for (int segment = 0; segment < numSegments; segment++) {
-            startTargetsPerSegment.add(startTarget);
-            final int numTargetsInSegment = numTargetsPerSegment.get(segment);
-            coveragesPerSegment.add(coverages.subList(startTarget, startTarget + numTargetsInSegment));
-            startTarget += numTargetsInSegment;
+        //partition coverages with target indices by segment
+        int targetIndex = 0;
+        for (final SimpleInterval segment : segments) {
```

Comment:
hmm, i think it doesn't hurt to keep the intermediates extracted for clarity.  for example, `targetCoverage.getCount()` might evoke an `int` quantity otherwise.  i'll take you up on inlining the increment, though!